setup.py

```diff
@@ -58,7 +58,7 @@ def run_tests(self):
     packages=find_packages(exclude=['tests*']),
     install_requires=[
         "wrapt",
-        "msgpack-python",
+        "msgpack",
```

Comment:
@k4nar after some investigation, I think there are some cases we should handle. I think depending on `msgpack` is something we must have since it will be the newer version + the one in roadmap to 1.0. Unfortunately it's possible that we're changing users implementation. For instance:
* we don't pin the `msgpack-python` version so that users may do in their applications (i.e. for specific reasons they have to use version 0.4.8)
* with the current `ddtrace` package, the version will stay 0.4.8 and we're fine with it (and so the user)
* we ship a new version that depends on `msgpack` that is version 0.5.4
* users update their dependency and so in their requirements file they have both `msgpack-python` and `msgpack`
* depending on the installation order, we may overwrite their version of msgpack that now is 0.5.4 and not 0.4.8 (and it could lead to unexpected behaviors in their application)

It's something temporarily that we will change on version 1.0, but until then we're "upgrading" the dependency in an unexpected way. Of course it applies on the other way round, but in the current state we'll at most set the same version (since 0.5.4 is available also for `msgpack-python`.

What do you think about this problem from your perspective?

Reply:
As msgpack is already an optional dependency of ddtrace (it works without it), maybe the best would be to provide it as an extra dependency.

If I want to install ddtrace with explicit msgpack support, I can run `pip install ddtrace[msgpack]`, else ddtrace will only use msgpack if it's available in my env.

Reply:
That's what `requests` is doing for SSL for example: https://github.com/requests/requests/blob/master/setup.py#L97-L100

Reply:
@k4nar actually `msgpack-python` is a mandatory dependency because without it the JSON encoder (and so the decoder in the Datadog Agent) is used, reducing the overall performance of the tracing system. JSON should be considered only a fallback when the C implementation for msgpack is not available.

What I'm saying is that JSON encoder/decoder doesn't make the tracing unusable; it's just that with msgpack it runs way better and so is highly recommended (and this is why we want to keep it as a dependency).

Reply:
I see only three options here:

1. Keep depending on `mgpack-python` and prevent users from upgrading msgpack
2. Depend on `msgpack` and break envs using older versions of `msgpack-python`
3. Make the dependency optional. State in the doc that you should run `pip install ddtrace[msgpack]`, unless you absolutly want to use the old `msgpack-python` package.

None of those options is perfect, but for me the N°1 is the worst, because it prevent users from moving forward. For example if there is a new release of `msgpack` with a security fix, I won't be able to use it because ddtrace will be using the old package.

Reply:
Solution (2) is the way to go sooner or later. What I'm saying is that it could be better to wait a bit more so that we probably don't break any env. For instance, a 0.6 or 1.0 release for `msgpack` (where `msgpack-python` will not be available) could be the right moment to switch to the new dependency and we can have time to notify our users. Solution (1) is definitely not a mid/long term solution because of what you said and I totally agree it's not the way to go.

Solution (2) and (3) may break current installations leading to unexpected behaviors (i.e. downgrading Tracer or Agent performances in services at scale). I think we can:
* wait to ship a new release (we have scheduled releases every 2/3 weeks)
* merge this change on `master`
* ship a release candidate with only this patch so people can test it and we can collect feedbacks
* ship a release after

Reply:
LGTM :+1:

Reply:
Great! I'm flagging this PR as "Request changes" just to keep it on hold (and leave a paper trail), though it's good exactly as it is now! Thank you very much for your precious feedback!